pyomo/contrib/alternative_solutions/obbt.py

```diff
@@ -0,0 +1,370 @@
+    )
+    new_constraint = False
+    if len(obj_constraints) > 0:
+        new_constraint = True
+    orig_objective.deactivate()
+
+    if use_appsi:
+        opt.update_config.check_for_new_or_removed_constraints = new_constraint
+        opt.update_config.check_for_new_or_removed_vars = False
+        opt.update_config.check_for_new_or_removed_params = False
+        opt.update_config.check_for_new_objective = True
+        opt.update_config.update_constraints = False
+        opt.update_config.update_vars = False
+        opt.update_config.update_params = False
+        opt.update_config.update_named_expressions = False
+        opt.update_config.update_objective = False
+        opt.update_config.treat_fixed_vars_as_params = False
+
+    variable_bounds = pe.ComponentMap()
+    solns = [Solution(model, all_variables, objective=orig_objective)]
+
+    senses = [(pe.minimize, "LB"), (pe.maximize, "UB")]
+
+    iteration = 1
+    total_iterations = len(senses) * num_vars
+    for idx in range(len(senses)):
+        sense = senses[idx][0]
+        bound_dir = senses[idx][1]
+
+        for var in variable_list:
+            if idx == 0:
+                variable_bounds[var] = [None, None]
+
+            if hasattr(aos_block, "var_objective"):
+                aos_block.del_component("var_objective")
+
+            aos_block.var_objective = pe.Objective(expr=var, sense=sense)
+
+            if warmstart:
+                _update_values(var, bound_dir, solutions)
+
+            if use_appsi:
+                opt.update_config.check_for_new_or_removed_constraints = new_constraint
+            if use_appsi:
+                opt.config.stream_solver = tee
+                results = opt.solve(model)
+                condition = results.termination_condition
+            else:
+                try:
+                    results = opt.solve(
+                        model, warmstart=warmstart, tee=tee, load_solutions=False
+                    )
+                except:
+                    results = None
+                if results is None:
+                    results = opt.solve(model, tee=tee, load_solutions=False)
+                condition = results.solver.termination_condition
+            new_constraint = False
+
+            if condition == optimal_tc:
+                if use_appsi:
+                    results.solution_loader.load_vars(solution_number=0)
+                else:
+                    model.solutions.load_from(results)
+                solns.append(Solution(model, all_variables, objective=orig_objective))
+
+                if warmstart:
+                    _add_solution(solutions)
+                obj_val = pe.value(var)
+                variable_bounds[var][idx] = obj_val
+
+                if refine_discrete_bounds and not var.is_continuous():
+                    if sense == pe.minimize and var.lb < obj_val:
+                        bound_name = var.name + "_" + str.lower(bound_dir)
```

Comment:
Are you sure this name will be unique? Also, why not just change the `bounds` on the `Var` object?

Reply:
We are only changing data in the AOS block to ensure that the model values don't get changed.

Within that block, this name will be unique if the variable names are unique.  I'm not sure that this will correctly handle variables on different blocks with the same name.  @jlgearh ???

Reply:
We discussed this and will switch to using the unique name function or changing the bounds and restoring them at the end.